storybookjs/storybook
# Next.js: Fix `useParams` support

Issue: https://github.com/storybookjs/storybook/issues/22077

## What I did

- Add `GlobalLayoutRouterContext` into `AppRouterProvider` to pass `useParams` validation in nextjs
- Add story for mocking `useParams`

## How to test

- Start a nextjs sandbox template
- Access http://localhost:6006/?path=/story/frameworks-nextjs-nextjs-default-js-navigation--with-segment-defined to see `useParams` usage doesn't throw any errors.
- Access http://localhost:6006/?path=/story/frameworks-nextjs-nextjs-default-js-navigation--with-segment-defined-for-params to see `useParams` mocking value

## Ref

[GitStart Ticket](https://clients.gitstart.com/storybook-oss/1/tickets/STORYBOOK-22077)
